### PR TITLE
Rename FirstOrderddt -> SecondOrderddt

### PR DESCRIPTION
**Pull request by schillic**

### CODE REVIEW

File: test/algorithms/LGG09.jl [hunk 1]
```diff
@@ -76,10 +76,10 @@ end
     prob, dt = foo()
     δ = 2e-1
 
-    approx_model = ReachabilityAnalysis.FirstOrderddt()
+    approx_model = ReachabilityAnalysis.SecondOrderddt()
```
Comment (mforets) [inline test/algorithms/LGG09.jl:79]:
```suggestion
    approx_model = SecondOrderddt()
```

Comment (mforets) [reply test/algorithms/LGG09.jl:79]: seems exported..

File: test/algorithms/LGG09.jl [hunk 2]
```diff
@@ -76,10 +76,10 @@ end
...
     sol = solve(prob, tspan=(0.0, 20.0),
                 alg=LGG09(δ=δ, template=PolarDirections(40), approx_model=approx_model))
-    approx_model = ReachabilityAnalysis.FirstOrderddt(oa=false)
+    approx_model = ReachabilityAnalysis.SecondOrderddt(oa=false)
```
Comment (mforets) [inline test/algorithms/LGG09.jl:82]:
```suggestion
    approx_model = SecondOrderddt(oa=false)
```

**Review by mforets — APPROVED** [review]
Do you recall if we renamed the method in the arxiv version of the review article as well?

**Comment by schillic** [comment]
> Do you recall if we renamed the method in the arxiv version of the review article as well?

We did not.

**Comment by schillic** [comment]
Should we merge this or do you want to sync?

**Comment by mforets** [comment]
I think we can merge this and sync later; i added https://github.com/JuliaReach/ReachabilityAnalysis.jl/issues/647 as a reminder.

**Pull request merged**